### PR TITLE
feat:  path construction

### PR DESCRIPTION
**Pull request by CarpentersKeys**
I found that these requires weren't working for me despite same relative path. This might cause fewer errors for people adapting the code.

Node.js v16.14.0.
discord.js  v13.6.0

**Please describe the changes this PR makes and why it should be merged:**

### CODE REVIEW

File: code-samples/creating-your-bot/command-handling/deploy-commands.js [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
+const path = require('path');
+const fs = require('fs');
 const { REST } = require('@discordjs/rest');
 const { Routes } = require('discord-api-types/v9');
 const { clientId, guildId, token } = require('./config.json');
 
 const commands = [];
-const commandFiles = fs.readdirSync('./commands').filter(file => file.endsWith('.js'));
+const commandsPath = path.join(__dirname, 'commands');
+const commandFiles = fs.readdirSync('commandsPath').filter(file => file.endsWith('.js'));
```
Comment (imranbarbhuiya) [inline code-samples/creating-your-bot/command-handling/deploy-commands.js:9]:
```suggestion
const commandFiles = fs.readdirSync(commandsPath).filter(file => file.endsWith('.js'));
```

**Review by imranbarbhuiya — CHANGES_REQUESTED** [review]

**Comment by imranbarbhuiya** [comment]
I just noticed it's duplicate of #1009

**Comment by almostSouji** [comment]
The dev PR'ing the original hasn't answered revision requests, so if you can look at https://github.com/discordjs/guide/pull/1009 and apply the requested changes @CarpentersKeys we can merge this instead, no problem.

File: code-samples/creating-your-bot/command-handling/deploy-commands.js [hunk 2]
```diff
@@ -1,10 +1,12 @@
-const fs = require('node:fs');
+const path = require('path');
+const fs = require('fs');
```
Comment (almostSouji) [inline code-samples/creating-your-bot/command-handling/deploy-commands.js:2]: keep fs in the node scope, please.

**Review by almostSouji — CHANGES_REQUESTED** [review]
This PR should also cover the other instances of code in the guide (code samples and explanations, where this is applicable).

File: code-samples/creating-your-bot/command-handling/deploy-commands.js (hunk 2 above)
Comment (CarpentersKeys) [reply code-samples/creating-your-bot/command-handling/deploy-commands.js:2]: may I convert modules to ESM?

Comment (megatank58) [reply code-samples/creating-your-bot/command-handling/deploy-commands.js:2]: The rest of the guide uses cjs based code, so should this section

File: guide/creating-your-bot/command-handling.md [hunk 3]
```diff
@@ -140,6 +140,7 @@ In your `index.js` file, make these additions:
 
 ```js {1-2,7}
 const fs = require('node:fs');
+const path = require('node:');
```
Comment (imranbarbhuiya) [inline guide/creating-your-bot/command-handling.md:143]:
```suggestion
const path = require('node:path');
```

File: guide/creating-your-bot/command-handling.md [hunk 4]
```diff
@@ -171,16 +177,19 @@ for (const file of commandFiles) {
 Use the same approach for your `deploy-commands.js` file, but instead `.push()` to the `commands` array with the JSON data for each command.
 
 ```js {1,7,9-12}
-const fs = require('node:fs');
+const fs = require('fs');
+const path = require('path');
```
Comment (imranbarbhuiya) [inline guide/creating-your-bot/command-handling.md:181]:
```suggestion
const fs = require('node:fs');
const path = require('node:path');
```

File: guide/creating-your-bot/event-handling.md [hunk 5]
```diff
@@ -71,12 +71,14 @@ Next, let's write the code for dynamically retrieving all the event files in the
 `fs.readdirSync().filter()` returns an array of all the file names in the given directory and filters for only `.js` files, i.e. `['ready.js', 'interactionCreate.js']`.
 
 ```js {3,5-12}
+const eventsPath = path.join(__dirname, 'events');
 const client = new Client({ intents: [Intents.FLAGS.GUILDS] });
```
Comment (imranbarbhuiya) [inline guide/creating-your-bot/event-handling.md:76]:
```suggestion
const client = new Client({ intents: [Intents.FLAGS.GUILDS] });

const eventsPath = path.join(__dirname, 'events');
```

**Review by imranbarbhuiya — CHANGES_REQUESTED** [review]

**Comment by CarpentersKeys** [comment]
Are there further changes required?

**Comment by megatank58** [comment]
> I could not get this to work with path.join(). I had to use fs.realpathSync() in order to avoid triggering an error. Additional note, __dirname does not resolve properly through my end. Results in the literal translation of attempting to find "__dirname" or "__localdir", as opposed to finding the representation.

can you provide the sample code with which the function failed

**Comment by Hades6578** [comment]
> > I could not get this to work with path.join(). I had to use fs.realpathSync() in order to avoid triggering an error. Additional note, __dirname does not resolve properly through my end. Results in the literal translation of attempting to find "__dirname" or "__localdir", as opposed to finding the representation.
> 
> can you provide the sample code with which the function failed

I simply used the code provided with the tutorial, I had hand-written the code first, and when it didn't work, copy/pasted the example from the repo.

File: code-samples/creating-your-bot/event-handling/index.js [hunk 6]
```diff
@@ -1,13 +1,17 @@
...
 
-const eventFiles = fs.readdirSync('./events').filter(file => file.endsWith('.js'));
+
+const eventsPath = path.join(__dirname, 'events');
+const eventFiles = fs.readdirSync(eventsPath).filter(file => file.endsWith('.js'));
 
 for (const file of eventFiles) {
-	const event = require(`./events/${file}`);
+	const filePath = path.join(eventsPath, file)
+	const event = require(filePath);
```
Comment (almostSouji) [inline code-samples/creating-your-bot/event-handling/index.js:14]:
```suggestion
	const filePath = path.join(eventsPath, file);
	const event = require(filePath);
```

**Review by almostSouji — APPROVED** [review]

**Pull request merged**